hackathon_site/event/test_api.py

```diff
@@ -193,3 +194,84 @@ def test_leave_with_not_empty_team(self):
         self.assertEqual(
             Team.objects.filter(team_code=old_team.team_code).exists(), True
         )
+
+
+class EventTeamListsViewTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.team2 = Team.objects.create()
+        self.team3 = Team.objects.create()
+
+        self.view = reverse("api:event:team-list")
+
+    def _login(self):
+        self.client.login(username=self.user.username, password=self.password)
+
+    def _login_permission(self):
+        self.client.login(username=self.user.username, password=self.password)
+        self.user.user_permissions.add(
+            Permission.objects.get(codename="view_team", pk="28")
```

Comment:
Getting the permission by its primary key (the codename argument doesn't do anything when you specify the pk) isn't great, since it could change if the order permissions are created at changes (that's a detail internal to django).

You can filter by content type to get the app name instead, which will be more robust:

```python
permissions = Permission.objects.get(content_type__app_label="event", codename="view_team")
```